configuration/components/loki.libsonnet

```diff
@@ -582,83 +618,43 @@ function(params) {
         '-target=' + normalizedName(component),
         '-config.file=/etc/loki/config/config.yaml',
         '-limits.per-user-override-config=/etc/loki/config/overrides.yaml',
-        '-log.level=error',
         // Necessary flag to support env variable expansion
         '-config.expand-env=true',
-      ] + (
-        if std.objectHas(osc, 'endpointKey') then [
-          '-s3.url=$(S3_URL)',
-          '-s3.force-path-style=true',
-        ] else [
-          '-s3.buckets=$(S3_BUCKETS)',
-          '-s3.region=$(S3_REGION)',
-          '-s3.access-key-id=$(AWS_ACCESS_KEY_ID)',
-          '-s3.secret-access-key=$(AWS_SECRET_ACCESS_KEY)',
-        ]
-      ) + (
-        if component == 'ruler' && std.objectHas(rsc, 'endpointKey') then [
-          '-ruler.storage.s3.url=$(RULER_S3_URL)',
-          '-ruler.storage.s3.force-path-style=true',
-        ] else if component == 'ruler' then [
-          '-ruler.storage.s3.buckets=$(RULER_S3_BUCKETS)',
-          '-ruler.storage.s3.region=$(RULER_S3_REGION)',
-          '-ruler.storage.s3.access-key-id=$(RULER_AWS_ACCESS_KEY_ID)',
-          '-ruler.storage.s3.secret-access-key=$(RULER_AWS_SECRET_ACCESS_KEY)',
-        ] else []
-      ),
+      ],
       // Some of the values present in here we only know them at rollout time
       // Loki config supports env variable expansion, we leverage this feature to
       // inject config parameters at rollout time.
       env: [
         // env only supports strings hence the addition with ''
         envVarFromValue('LOKI_REPLICATION_FACTOR', '' + loki.config.replicationFactor),
+        envVarFromValue('LOKI_LOG_LEVEL', loki.config.logLevel),
       ] + (
+        if component == 'querier' then [
+          // env only supports strings hence the addition with ''
+          envVarFromValue('LOKI_QUERIER_MAX_CONCURRENCY', '' + loki.config.query.concurrency),
+        ] else []
+      ) + (
+        if component == 'ingester' then [
+          // env only supports strings hence the addition with ''
+          envVarFromValue('LOKI_INGESTER_WAL_REPLAY_MEMORY_CEILING', '' + loki.config.wal.replayMemoryCeiling),
+        ] else []
+      ) + (
```

Comment:
@periklis I want to double-check something with you. I'm afraid that if we do things like this then components that don't have a given env var will not correctly perform ENV expansion and will have a problem parsing the configuration, the CI is green so apparently this is not an issue but I wanted your opinion.

One solution if this is indeed a problem is to provide default values which AFAIK the loki config supports
https://grafana.com/docs/loki/latest/configuration/#use-environment-variables-in-the-configuration

Reply:
Me and Peri talked about it and decided to simply add the ENV vars to all the components deployments/statefulsets